[question] (Conan 2.0) on MSVC `conan profile detect` leads to runtime conflicts when using `-s build_type=Debug`

### What is your question?

Using the default settings from `conan profile detect` for Windows.

Will give you the following on `conan install`

```
======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release 👈 This here is populated for both profiles
compiler.version=193
os=Windows

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows
```

However the default `runtime_type` profile plugin

```py
    settings = profile.settings
    if settings.get("compiler") == "msvc" and settings.get("compiler.runtime"):
        if settings.get("compiler.runtime_type") is None: # 👈  This only kicks in if the setting is missing -- which it never is
            runtime = "Debug" if settings.get("build_type") == "Debug" else "Release"
            try:
                settings["compiler.runtime_type"] = runtime
            except ConanException:
                pass
```

So the more simple use case `conan install -s build_type=Debug` leads to the profile helper not kicking in and mismatched sysmbols

```
======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Debug 👈  This is the easy way
compiler=msvc
compiler.cppstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release 🚫 Does not match
compiler.version=193
os=Windows

Profile build:
[settings]
arch=x86_64
build_type=Debug
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows
```

So it will result in 

```
Catch2d.lib(catch_session.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in Main.obj [D:\a\argon\argon\build\tests\test-argon.vcxproj]
Catch2d.lib(catch_session.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in Main.obj [D:\a\argon\argon\build\tests\test-argon.vcxproj]
```

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

--- a/conans/client/conf/detect.py
+++ b/conans/client/conf/detect.py
@@ -224,7 +224,6 @@ def _detect_compiler_version(result):
         # Add default mandatory fields for MSVC compiler
         result.append(("compiler.cppstd", "14"))
         result.append(("compiler.runtime", "dynamic"))
-        result.append(("compiler.runtime_type", "Release"))
 
     if compiler != "msvc":
         cppstd = _cppstd_default(compiler, version)

--- a/conans/test/functional/command/profile_test.py
+++ b/conans/test/functional/command/profile_test.py
@@ -10,7 +10,7 @@ from conans.client.conf.detect import detect_defaults_settings
 from conans.test.utils.mocks import RedirectedTestOutput
 from conans.test.utils.tools import TestClient, redirect_output
 from conans.util.env import environment_update
-from conans.util.files import load, save
+from conans.util.files import save
 from conans.util.runners import check_output_runner
 
 
@@ -118,12 +118,13 @@ class DetectCompilersTest(unittest.TestCase):
         self.assertIn("gcc detected as a frontend using apple-clang", output)
 
     def test_profile_new(self):
-        client = TestClient()
+        c = TestClient()
+        c.run("profile detect --name=./MyProfile2")
+        profile = c.load("MyProfile2")
+        assert "os=" in profile
+        assert "compiler.runtime_type" not in profile  # Even in Windows
 
-        client.run("profile detect --name=./MyProfile2")
-        pr_path = os.path.join(client.current_folder, "MyProfile2")
-        self.assertTrue(os.path.exists(os.path.join(client.current_folder, "MyProfile2")))
-        self.assertIn("os=", load(pr_path))
+        c.run("profile detect --name=./MyProfile2", assert_error=True)
+        assert "MyProfile2' already exists" in c.out
 
-        client.run("profile detect --name=./MyProfile2", assert_error=True)
-        self.assertIn("MyProfile2' already exists", client.out)
+        c.run("profile detect --name=./MyProfile2 --force")  # will not raise error

